changelog/changes/2023-10-25-docker-gentoo-upstream.md

```diff
@@ -0,0 +1,4 @@
+- cri-tools, runc, containerd, docker, and docker-cli are now shipped without debugging symbols and built from Gentoo upstream ebuilds. Docker was updated to Docker 24 (see "updates").
+  - **NOTE** The docker btrfs storage driver has been de-prioritised; BTRFS backed storage will now default to the `overlay2` driver
+    ([changelog](https://docs.docker.com/engine/release-notes/23.0/#bug-fixes-and-enhancements-6), [upstream pr](https://github.com/moby/moby/pull/42661)).
+    Using the btrfs driver can still be enforced by creating a respective [docker config](https://docs.docker.com/storage/storagedriver/btrfs-driver/#configure-docker-to-use-the-btrfs-storage-driver) at `/etc/docker/daemon.json`.
```

Comment:
Does this affect existing installations or is it only for the new ones? Maybe it's something to add to the changelog entry?

Reply:
It would affect existing deployments that use Docker's brtfs storage driver. I've added a more explicit note to call this out.

Reply:
Good news: Docker includes a backwards compatibility mode for btrfs storage: If the storage is actively used, docker will automatically select the `btrfs` storage driver. I've added a [Mantle test](https://github.com/flatcar/mantle/pull/475) to validate that.